packages/model-generator/API.md

```diff
@@ -150,10 +165,10 @@ export type GraphqlModelsGenerator = {
 // @public (undocumented)
 export type GraphqlModelsGeneratorFactoryParams = {
     backendIdentifier: DeployedBackendIdentifier;
-    credentialProvider: AwsCredentialIdentityProvider;
+    options: GraphqlModelsFetchOptions;

```

Comment:
1. can we flatten this?
2. should options be optional ? I.e. fall back to default client/credential provider if not specified ?

Reply:
I'm not sure what you mean about flattening it. I have a similar discussion with @Amplifiyer above (https://github.com/aws-amplify/amplify-backend/pull/1288#discussion_r1565976866), but I think it's better to group the alternative sdk mechanisms (clients vs credentialProvider) into one object.

Reply:
We should continue discussion about whether to make `options` optional here: https://github.com/aws-amplify/amplify-backend/pull/1288#discussion_r1565984937

Reply:
> I'm not sure what you mean about flattening it

Instead of having `params.options.s3Client` have `params.s3Client` which is optional and mutually exclusive with `params.credentialsProvider` (which is also optional).

Reply:
but if intent is to group SDK related options then it needs more precise naming.
`params.options` doesn't not look good. `params.clientConfiguration` might be better.

Reply:
@sobolk what do you think about this https://github.com/aws-amplify/amplify-backend/pull/1288#discussion_r1566036600

Reply:
I think it's a good alternative too. and removes a lot of type algebra.
If we do this we should have `ClientProvider` as optional param. 
I would however avoid putting all factory methods into single interface and have `ClientProvider<S3Client>`, `ClientProvider<CloudFormationClient>` separately. Otherwise we'll be dragging whole SDK whenever we use interface.